Add rpm package build script

kiichi7 opened the pull request:
I added RPM package build script for SLES/open SUSE and RHEL/Cent OS.

zonyitoo commented:
@kiichi7 Do you still want to push this PR forward? I don't have any experience in RPM packaging.

kiichi7 commented:
I have use the same rpm build script to build shadowsock-rust.x86_64-<VERSION>.rpm for many years.

> sudo yum --showduplicates list shadowsocks-rust                                                                                                                                                                  
[sudo] password for sysadm:                                                                                                                                                                                        
Loaded plugins: product-id, search-disabled-repos, subscription-manager, versionlock                                                                                                                               
Excluding 1 update due to versionlock (use "yum versionlock status" to show it)                                                                                                                                    
Available Packages                                                                                                                                                                                                 
shadowsocks-rust.x86_64                                              1.6.0-1.el7                                                 shadowsocks                                                                       
shadowsocks-rust.x86_64                                              1.6.1-1.el7                                                 shadowsocks                                                                       
shadowsocks-rust.x86_64                                              1.6.2-1.el7                                                 shadowsocks                                                                       
shadowsocks-rust.x86_64                                              1.6.3-1.el7                                                 shadowsocks                                                                       
shadowsocks-rust.x86_64                                              1.6.4-1.el7                                                 shadowsocks                                                                       
shadowsocks-rust.x86_64                                              1.6.5-1.el7                                                 shadowsocks                                                                       
shadowsocks-rust.x86_64                                              1.6.6-1.el7                                                 shadowsocks                                                                       
shadowsocks-rust.x86_64                                              1.6.7-1.el7                                                 shadowsocks                                                                       
shadowsocks-rust.x86_64                                              1.6.8-1.el7                                                 shadowsocks                                                                       
shadowsocks-rust.x86_64                                              1.6.9-1.el7                                                 shadowsocks                                                                       
shadowsocks-rust.x86_64                                              1.6.10-1.el7                                                shadowsocks                                                                       
shadowsocks-rust.x86_64                                              1.6.11-1.el7                                                shadowsocks                                                                       
shadowsocks-rust.x86_64                                              1.6.12-1.el7                                                shadowsocks                                                                       
shadowsocks-rust.x86_64                                              1.7.0-1.el7                                                 shadowsocks                                                                       
shadowsocks-rust.x86_64                                              1.7.2-1.el7_9                                               shadowsocks                                                                       
shadowsocks-rust.x86_64                                              1.8.0-1.el7_9                                               shadowsocks                                                                       
shadowsocks-rust.x86_64                                              1.8.1-1.el7_9                                               shadowsocks                                                                       
shadowsocks-rust.x86_64                                              1.8.2-1.el7_9                                               shadowsocks                                                                       
shadowsocks-rust.x86_64                                              1.8.3-1.el7_9                                               shadowsocks                                                                       
shadowsocks-rust.x86_64                                              1.8.4-1.el7_9                                               shadowsocks                                                                       
shadowsocks-rust.x86_64                                              1.8.5-1.el7_9                                               shadowsocks                                                                       
shadowsocks-rust.x86_64                                              1.8.6-1.el7_9                                               shadowsocks                                                                       
shadowsocks-rust.x86_64                                              1.8.7-1.el7_9                                               shadowsocks                                                                       
shadowsocks-rust.x86_64                                              1.8.8-1.el7_9                                               shadowsocks                                                                       
shadowsocks-rust.x86_64                                              1.8.9-1.el7_9                                               shadowsocks                                                                       
shadowsocks-rust.x86_64                                              1.8.10-1.el7_9                                              shadowsocks                                                                       
shadowsocks-rust.x86_64                                              1.8.11-1.el7_9                                              shadowsocks                                                                       
shadowsocks-rust.x86_64                                              1.8.12-1.el7_9                                              shadowsocks                                                                       
shadowsocks-rust.x86_64                                              1.8.13-1.el7_9                                              shadowsocks                                                                       
shadowsocks-rust.x86_64                                              1.8.14-1.el7_9                                              shadowsocks                                                                       
shadowsocks-rust.x86_64                                              1.8.15-1.el7_9                                              shadowsocks                                                                       
shadowsocks-rust.x86_64                                              1.8.16-1.el7_9                                              shadowsocks                                                                       
shadowsocks-rust.x86_64                                              1.8.17-1.el7_9                                              shadowsocks                                                                       
shadowsocks-rust.x86_64                                              1.8.18-1.el7_9                                              shadowsocks                                                                       
shadowsocks-rust.x86_64                                              1.8.19-1.el7_9                                              shadowsocks                                                                       
shadowsocks-rust.x86_64                                              1.8.20-1.el7_9                                              shadowsocks                                                                       
shadowsocks-rust.x86_64                                              1.8.21-1.el7_9                                              shadowsocks                                                                       
shadowsocks-rust.x86_64                                              1.8.22-1.el7_9                                              shadowsocks                                                                       
shadowsocks-rust.x86_64                                              1.8.23-1.el7_9                                              shadowsocks                                                                       
shadowsocks-rust.x86_64                                              1.9.0-1.el7_9                                               shadowsocks                                                                       
shadowsocks-rust.x86_64                                              1.9.1-1.el7_9                                               shadowsocks                                                                       
shadowsocks-rust.x86_64                                              1.9.2-1.el7_9                                               shadowsocks                                                                       
shadowsocks-rust.x86_64                                              1.10.0-1.el7_9                                              shadowsocks                                                                       
shadowsocks-rust.x86_64                                              1.10.1-1.el7_9                                              shadowsocks                                                                       
shadowsocks-rust.x86_64                                              1.10.2-1.el7_9                                              shadowsocks                                                                       
shadowsocks-rust.x86_64                                              1.10.3-1.el7_9                                              shadowsocks                                                                       
shadowsocks-rust.x86_64                                              1.10.4-1.el7_9                                              shadowsocks                                                                       
shadowsocks-rust.x86_64                                              1.10.5-1.el7_9                                              shadowsocks                                                                       
shadowsocks-rust.x86_64                                              1.10.6-1.el7_9                                              shadowsocks

you can check on this repo URL `baseurl=https://www.kiichi7.com/packages/stable`

kiichi7 commented:
I removed the build dependencies and package dedendencies for the rust version, and updated the init.d / systemd deamon files for rust version. 
For files in /scripts/  I  will clean up.
